extensions/amp-onetap-google/0.1/amp-onetap-google.js

```diff
@@ -0,0 +1,367 @@
+  border: 'none',
+  margin: 0,
+  padding: 0,
+  position: 'fixed',
+  right: 'auto',
+  left: '0',
+  top: 'auto',
+  bottom: '0',
+  width: '100%',
+};
+
+/** @const {Object} */
+const elementStyle = {
+  height: '0',
+  width: '0',
+  'z-index': 2147483647,
+  border: 'none',
+  position: 'fixed',
+  left: '0',
+  bottom: '0',
+};
+
+export class AmpOnetapGoogle extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    this.checkExperimentEnabled_();
+
+    /** @private @const {function} */
+    this.doneCallback_ = this.getDoneCallback(
+      this.element.getAttribute(ATTRIBUTE_DONE_CALLBACK)
+    );
+
+    /** @private @const {string} */
+    this.intermediateIframeUrl_ = assertHttpsUrl(
+      this.element.getAttribute(ATTRIBUTE_IFRAME_URL),
+      ATTRIBUTE_IFRAME_URL
+    );
+
+    /** @private @const {string} */
+    this.intermediateIframeOrigin_ = new URL(
+      this.intermediateIframeUrl_
+    ).origin;
+
+    /** @private {!Element} */
+    this.doc_ = this.element.ownerDocument;
+
+    /** @private @const {!../../../src/service/vsync-impl.Vsync} */
+    this.vsync_ = Services.vsyncFor(
+      toWin(this.element.ownerDocument.defaultView)
+    );
+
+    /** @private {?boolean} */
+    this.isBottomSheetUiMode_ = null;
+
+    /** @private {?Element} */
+    this.intermediateIframe_ = null;
+
+    /** @private {boolean} */
+    this.isIntermediateIframeVisible_ = false;
+
+    /** @private {boolean} */
+    this.cancelOnTapOutside_ = true;
+
+    /** @private */
+    this.onClickOutside_ = () => {
+      if (this.cancelOnTapOutside_ && this.isIntermediateIframeVisible_) {
+        this.doc_.removeEventListener('click', this.onClickOutside_);
+        this.removeIntermediateIframe_();
+      }
+    };
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.NODISPLAY;
+  }
+
+  /**
+   * Parses the callback on One Tap flow done.
+   * @param {string=} callbackName
+   * @return {function(...[*]=)}
+   */
+  getDoneCallback(callbackName) {
+    if (callbackName && window[callbackName]) {
+      const cb = window[callbackName];
+      if (typeof cb === 'function') {
+        // found a valid done callback
+        return cb;
+      }
+    }
+    return () => {
+      window.location.reload();
+    };
+  }
+
+  /** @override */
+  buildCallback() {
+    this.win.addEventListener('message', (event) => {
+      // Make sure the postMessage comes from the iframe origin
+      if (this.validateMessageOrigin_(event.origin)) {
+        this.handleIntermediateIframeMessage_(event);
+      }
+    });
+    if (document.readyState === 'complete') {
+      // window Load event has already been fired. Use set timeout to defer
+      // the call.
+      setTimeout(() => {
+        this.loadIntermediateIframe_();
+      }, 200);
+    } else {
+      this.win.addEventListener('load', function () {
+        this.loadIntermediateIframe_();
+      });
+    }
```

Comment:
To keep privacy guarantees:

```suggestion
    this.ampdoc.whenFirstVisible().then(() => {
      this.loadIntermediateIframe_();
    });
```

Reply:
Done.